cpp/tests/community/k_truss_test.cu

```diff
@@ -0,0 +1,408 @@
+
+        thrust::sort(handle.get_thrust_policy(), d_edge_first, d_edge_first + d_cugraph_src.size());
+      } else {
+        auto d_edge_first = thrust::make_zip_iterator(
+          thrust::make_tuple(d_cugraph_src.begin(), d_cugraph_dst.begin()));
+
+        thrust::sort(handle.get_thrust_policy(), d_edge_first, d_edge_first + d_cugraph_src.size());
+      }
+
+      EXPECT_EQ(d_cugraph_src.size(), d_reference_src.size());
+
+      ASSERT_TRUE(thrust::equal(handle.get_thrust_policy(),
+                                d_cugraph_src.begin(),
+                                d_cugraph_src.end(),
+                                d_reference_src.begin()));
+
+      ASSERT_TRUE(thrust::equal(handle.get_thrust_policy(),
+                                d_cugraph_dst.begin(),
+                                d_cugraph_dst.end(),
+                                d_reference_dst.begin()));
+
+      if (edge_weight_view) {
+        auto compare_functor = cugraph::test::device_nearly_equal<weight_t>{
+          weight_t{1e-3},
+          weight_t{(weight_t{1} / static_cast<weight_t>((*d_cugraph_wgt).size())) *
+                   weight_t{1e-3}}};
+        EXPECT_TRUE(thrust::equal(handle.get_thrust_policy(),
+                                  (*d_cugraph_wgt).begin(),
+                                  (*d_cugraph_wgt).end(),
+                                  (*d_reference_wgt).begin(),
+                                  compare_functor));
+      }
+    }
+  }
+};
+
+using Tests_KTruss_File = Tests_KTruss<cugraph::test::File_Usecase>;
+using Tests_KTruss_Rmat = Tests_KTruss<cugraph::test::Rmat_Usecase>;
+
+TEST_P(Tests_KTruss_File, CheckInt32Int32Float)
+{
+  run_current_test<int32_t, int32_t, float>(
+    override_File_Usecase_with_cmd_line_arguments(GetParam()));
+}
+
+TEST_P(Tests_KTruss_File, CheckInt64Int64Float)
+{
+  run_current_test<int64_t, int64_t, float>(
+    override_File_Usecase_with_cmd_line_arguments(GetParam()));
+}
+
+TEST_P(Tests_KTruss_File, CheckInt32Int64Float)
+{
+  run_current_test<int32_t, int64_t, float>(
+    override_File_Usecase_with_cmd_line_arguments(GetParam()));
+}
+
+TEST_P(Tests_KTruss_Rmat, CheckInt32Int32Float)
+{
+  run_current_test<int32_t, int32_t, float>(
+    override_Rmat_Usecase_with_cmd_line_arguments(GetParam()));
+}
+
+TEST_P(Tests_KTruss_Rmat, CheckInt64Int64Float)
+{
+  run_current_test<int64_t, int64_t, float>(
+    override_Rmat_Usecase_with_cmd_line_arguments(GetParam()));
+}
+
+TEST_P(Tests_KTruss_Rmat, CheckInt32Int64Float)
+{
+  run_current_test<int32_t, int64_t, float>(
+    override_Rmat_Usecase_with_cmd_line_arguments(GetParam()));
+}
+
+INSTANTIATE_TEST_SUITE_P(
+  simple_test,
+  Tests_KTruss_File,
+  ::testing::Combine(
+    // enable correctness checks
+    ::testing::Values(KTruss_Usecase{5, true, false},
+                      KTruss_Usecase{4, true, false},
+                      KTruss_Usecase{9, true, true},
+                      KTruss_Usecase{7, true, true}),
```

Comment:
If `check_correctness` is the last member variable with the default value of true, we can just provide `k` and `test_weighted` for non-benchmark tests.